airflow/providers/common/sql/operators/sql.py

```diff
@@ -51,6 +51,36 @@ def _parse_boolean(val: str) -> str | bool:
     raise ValueError(f"{val!r} is not a boolean-like string value")
 
 
+def _get_failed_checks(checks, col=None):
+    """
+    IMPORTANT!!! Keep it for compatibility with released 8.4.0 version of google provider.
+
+    Unfortunately the provider used _get_failed_checks and parse_boolean as imports and we should
+    keep those methods to avoid 8.4.0 version from failing.
+    """
+    if col:
+        return [
+            f"Column: {col}\nCheck: {check},\nCheck Values: {check_values}\n"
+            for check, check_values in checks.items()
+            if not check_values["success"]
+        ]
+    return [
+        f"\tCheck: {check},\n\tCheck Values: {check_values}\n"
+        for check, check_values in checks.items()
+        if not check_values["success"]
+    ]
+
+
+def parse_boolean(val: str) -> str | bool:
+    """
+    IMPORTANT!!! Keep it for compatibility with released 8.4.0 version of google provider.
```

Comment:
Keep untill when? What is the condition that it will be safe to remov

Reply:
Good question. I think there is no obvious removal condition. I could not find a good one. 8.4.0 can be kept for a long time and and one can continue upgrading common sql. you can easily install 8.4.0 on Airflow 2.8 and with comon.sql 1.10 for example - theorethically should be no problem. We have no rules for deprecating / EOL of providers. themselves.

Reply:
This is one of the problems with such a common code. You HAVE TO keep it as backwards compatible as possible - because it is shared between multiple packages., You can very easily imagine someone sticking with google provider  8.4.0 and upgrading some other providers that are using common.sql

This is somethign I mentioned at the beginning of our common.sql journey - once we released something there, it should stay there basically forever (unless we introduce EOL policies for providers, which we do not have currently).

Reply:
So why not releasing 8.4.1 with limiting the sql provider. It wont solve the issue but it can help users to quickly migrate out of 8.4.0 as nothing is changed other than pinning upper version which is also the version they are currently using

Reply:
Because it has been already installed and running for over a month and upgrading conmon.sql provider will break it now. and the worst thing is that by upgrading another package you break google one. This is quite unexpected for those users and will create a flurry of problems if we do not act quickly. 

I think it's a small price to keep 20 lines of code.

Reply:
Note - we have absolutely no mechanism to move people out of 8.4.0 if they already installed it. Wile common-sql 1.3.0 is not installe by almost anyone.

Reply:
Another variation. I think we have two different threats that we want to handle with single solution but they can be done separately - tactical problem of blocking immediate threat of our users getting hit by unexpected bigquery problem and strategic problem of long term deprecated code removal. 

Both problems can be solved by two paralell threads - there is no need to solve both at once.

I propose:

* we release 1.3.1 common.sql now (tomorrow) with my fix (tactical solution)
* you prepare google 8.4.1 release with "proper" usage of common.sql (rtght after)  - branch created from 8.4.0 tag - happy to guide you and review
* once released 8.4.1 we agree a suitable condition on when the code is removed and we update it in main. I would love to hear some good criteria for that proposed by you
* once the condition is fulfilled, we remove the re-added code (strategic solution).

I think that solves both problems independently. Tactical can be done now, strategic can be dofferred and applied later as next step.

We also start sharing more and more release management of providers (which is also my goal).

WDYT @eladkal ?

Reply:
We need this PR anyway so lets merge it and prepare the first step. we can discuss additional steps in parallel/after

Reply:
rather than adding the code back, it might be better to instead just add it to the google provider and make a patch release for that.

if ultimately we don't want it in core, seems best to keep it out. wdyt?

Reply:
@dstandish that will not solve the issue.
The issue is that google provider 8.4.0 which was released a while ago had no upper limit for sql provider. So once we release new version of sql provider we are breaking all installs with 8.4.0

There is no issue for people who updated google provider for new version. Version 8.5.0 works fine with the new sql provider release.

Reply:
oh, yeah

Reply:
Yes. The big problem is that the cat is out of the bag now. And just upgrading to currently released 1.3.0 of common.sql will break already released 8.4.0 - in a bad way so releasing 1.3.1 is just prevention of likely problems to happen.

1) release 1.3.1 QUICKLY and yank 1.3.0  (the coming wave of provider fixes) - result - no matter what combo of common.sql + google provider user has, it will work. This is plain, tactical solution to prevent flood of issues to handle and big inconvenience for our users. I'd call it "preventive hotfix".

2) reease 8.4.1 with a fix as next step.

3) decide what will be the condition to remove the added code in common.sql